pkg/arvo/app/chat-cli.hoon

```diff
@@ -0,0 +1,1111 @@
+    ::
+    ++  turl
+      =-  (sear - text)
+      |=  t=cord
+      ^-  (unit cord)
+      ?~((rush t aurf:de-purl:html) ~ `t)
+    ::  +text: text message body
+    ::
+    ++  text
+      %+  cook  crip
+      (plus ;~(less (jest '•') next))
+    ::  +expr: parse expression into [cord hoon]
+    ::
+    ++  expr
+      |=  tub=nail
+      %.  tub
+      %+  stag  (crip q.tub)
+      wide:(vang & [&1:% &2:% (scot %da now.bowl) |3:%])
+    --
+  ::  +obey: apply result
+  ::
+  ::    called upon hitting return in the prompt.
+  ::    if input is invalid, +slug is called.
+  ::    otherwise, the appropriate work is done and
+  ::    the command (if any) gets echoed to the user.
+  ::
+  ++  obey
+    ^-  (quip move _this)
+    =+  buf=buf.state.cli
+    =+  fix=(sanity [%nop ~] buf)
+    ?^  lit.fix
+      (slug fix)
+    =+  jub=(rust (tufa buf) read)
+    ?~  jub  [[(effect:sh-out %bel ~) ~] this]
+    =^  cal  state.cli  (~(transmit sole-lib state.cli) [%set ~])
+    =^  moves  this  (work u.jub)
+    :_  this
+    %+  weld
+      ^-  (list move)
+      ::  echo commands into scrollback
+      ?.  =(`0 (find ";" buf))  ~
+      [(note:sh-out (tufa `(list @)`buf)) ~]
+    :_  moves
+    %+  effect:sh-out  %mor
+    :~  [%nex ~]
+        [%det cal]
+    ==
+  ::  +work: run user command
+  ::
+  ++  work
+    |=  job=command
+    ^-  (quip move _this)
+    |^  ?-  -.job
+          %target    (set-target +.job)
+          %say       (say +.job)
+          %eval      (eval +.job)
+        ::
+          %join      (join +.job)
+          %leave     (leave +.job)
+          %create    (create +.job)
+        ::
+          %bind      (bind-glyph +.job)
+          %unbind    (unbind-glyph +.job)
+          %what      (lookup-glyph +.job)
+        ::
+          %settings  show-settings
+          %set       (set-setting +.job)
+          %unset     (unset-setting +.job)
+          %width     (set-width +.job)
+          %timezone  (set-timezone +.job)
+        ::
+          %select    (select +.job)
+          %chats     chats
+          %help      help
+        ==
+    ::  +act: build action move
+    ::
+    ++  act
+      |=  [what=term app=term =out-action]
+      ^-  move
+      :*  ost.bowl
+          %poke
+          /cli-command/[what]
+          [our-self app]
+          out-action
+      ==
+    ::  +set-target: set audience, update prompt
+    ::
+    ++  set-target
+      |=  tars=(set target)
+      ^-  (quip move _this)
+      =.  audience  tars
+      [[prompt:sh-out ~] this]
+    ::  +create: new local mailbox
+    ::
+    ++  create
+      ::TODO  configurable security
+      |=  [=path gyf=(unit char)]
+      ^-  (quip move _this)
+      ::TODO  check if already exists
+      =/  =target  [our-self path]
+      =^  moz  this
+        ?.  ?=(^ gyf)  [~ this]
+        (bind-glyph u.gyf target)
+      =-  [[- moz] this(audience [target ~ ~])]
+      %^  act  %do-create  %chat-view
+      :-  %chat-view-action
```

Comment:
Great. Glad you're using %chat-view-action here. It would be easy for you to support more security types.